python/pyiceberg/expressions/base.py

```diff
@@ -131,11 +75,115 @@ def __ge__(self, other):
 
 
 class BooleanExpression(ABC):
-    """base class for all boolean expressions"""
+    """Represents a boolean expression tree."""
 
     @abstractmethod
     def __invert__(self) -> "BooleanExpression":
-        ...
+        """Transform the Expression into its negated version."""
+
+
+class Bound(Generic[T], ABC):
+    """Represents a bound value expression."""
+
+    def eval(self, struct: StructProtocol):  # pylint: disable=W0613
+        ...  # pragma: no cover
+
+
+class Unbound(Generic[T, B], ABC):
+    """Represents an unbound expression node."""
+
+    @abstractmethod
+    def bind(self, schema: Schema, case_sensitive: bool) -> B:
+        ...  # pragma: no cover
+
+
+class Term(ABC):
+    """An expression that evaluates to a value."""
```

Comment:
I know this is from the [javadoc](https://iceberg.apache.org/javadoc/latest/org/apache/iceberg/expressions/Term.html), but is the word "expression" in this docstring a little confusing? Currently a "term" is either a reference or a transform of a reference. Should we just be explicit here and says `"A transform or reference that evaluates to a value."`?